modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -0,0 +1,288 @@
+      const userKeychain = await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+      const backupKeychain = await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+      userKeychain.should.deepEqual(nockedUserKeychain);
+
+      // unencrypted `prv` property should exist on backup keychain
+      const keyChainPrv = JSON.parse(backupKeychain.prv ?? '');
+      _.isEqual(keyChainPrv.pShare, backupKeyShare.pShare).should.be.true();
+      _.isEqual(keyChainPrv.bitgoNShare, bitgoKeyShare.nShares[2]).should.be.true();
+      _.isEqual(keyChainPrv.userNShare, userKeyShare.nShares[2]).should.be.true();
+      should.exist(backupKeychain.encryptedPrv);
+    });
+
+    it('should generate TSS key chains with optional params', async function () {
+      const enterprise = 'enterprise';
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const nockedUserKeychain = await nockUserKeychain({ coin: coinName });
+      await nockBackupKeychain({ coin: coinName });
+
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare, enterprise);
+      const userKeychain = await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase',
+        'originalPasscodeEncryptionCode');
+      const backupKeychain = await tssUtils.createBackupKeychain(
+        userGpgKey,
+        userKeyShare,
+        backupKeyShare,
+        bitgoKeychain,
+        'passphrase');
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+      userKeychain.should.deepEqual(nockedUserKeychain);
+
+      // unencrypted `prv` property should exist on backup keychain
+      const keyChainPrv = JSON.parse(backupKeychain.prv ?? '');
+      _.isEqual(keyChainPrv.pShare, backupKeyShare.pShare).should.be.true();
+      _.isEqual(keyChainPrv.bitgoNShare, bitgoKeyShare.nShares[2]).should.be.true();
+      _.isEqual(keyChainPrv.userNShare, userKeyShare.nShares[2]).should.be.true();
+      should.exist(backupKeychain.encryptedPrv);
+    });
+
+    it('should fail to generate TSS key chains', async function () {
+      const nockedBitGoKeychain = await nockBitgoKeychain({
+        coin: coinName,
+        userKeyShare,
+        backupKeyShare,
+        userGpgKey,
+      });
+      const bitgoKeychain = await tssUtils.createBitgoKeychain(userGpgKey, userKeyShare, backupKeyShare);
+      bitgoKeychain.should.deepEqual(nockedBitGoKeychain);
+
+      await tssUtils.createUserKeychain(
+        userGpgKey,
+        userKeyShare,
+        await MPC.keyShare(2, 2, 3),
```

Comment:
or even better, since we are only checking for difference in common key chain, instead of generating a new one, make a random modification in common keychain for an existing keyshares from before hook and use it here